### DISCUSSION TITLE
[2-2] 동적할당 해제

### DISCUSSION THREAD

**Discussion by SeoeunY** [bbang-jun/2023-OOP-Mentoring#11] [Q&A]
![image](https://user-images.githubusercontent.com/108859207/216375487-cb7254f5-5db5-4022-a220-d1785571988f.png)

아래 부분의 소멸자 부분에서 오류가 나서 질문드립니다. 동적할당 해제를 어떻게 해야될까요? 주석처리하면 오류가 안나는데 주석처리를 안하면 예외가 발생한다고 뜹니다.

**Comment by Bjimin** [comment]
일정이 있어서 늦게 답변드리게 되었네요 죄송합니다
일단 저런 방식으로 메모리를 해제하는 경우는 한 번에 여러개의 동적할당을 하는 경우, 즉 아래와 같은 코드로 동적할당 하는 경우에 해당합니다.
animal *animal_list = new animal[100];
-(위 코드는 길이가 100인 일차원 배열을 동적할당 하는 경우임)

하지만 위 코드를 보면 'animal class 형태 데이터의 주소'를 저장할 수 있는 길이 100의 배열을 먼저 선언하고, (이때 동적할당은 아직 x) 이후 각 칸에 animal class 형태로 하나씩 동적할당하여 데이터를 저장하는것이기 때문에 각각에 대해 메모리를 해제해줘야 오류가 없을 것으로 보입니다. 
직접 돌려보지 못해서 혹시 문제있으면 추가질문 부탁드립니다.
